ntp-daemon/src/main.rs

```diff
@@ -23,5 +27,66 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // this away when not using sentry.
     let _guard = finish_tracing_init(&mut config, has_log_override)?;
 
-    ntp_daemon::spawn(&config.system, &config.peers).await
+    let peers_reader = Arc::new(tokio::sync::RwLock::new(Vec::new()));
+    let peers_writer = peers_reader.clone();
+
+    std::fs::remove_dir_all("/run/ntpd-rs")?;
+    std::fs::create_dir("/run/ntpd-rs")?;
+
+    let peers_listener = UnixListener::bind("/run/ntpd-rs/observe")?;
+
+    // this binary needs to run as root to be able to adjust the system clock
+    // directories inherit root permissions, but the client should not need elevated
+    // permissions to read from the socket. So we explicitly set the permissions
+    let permissions: std::fs::Permissions = PermissionsExt::from_mode(0o777);
+
+    std::fs::set_permissions("/run/ntpd-rs", permissions.clone())?;
+    std::fs::set_permissions("/run/ntpd-rs/observe", permissions)?;
```

Comment:
we assume we're the only `ntpd-rs` instance that is running. That seems like a fair assumption to me

Reply:
agree that that assumption is reasonable, we need it elsewhere as well (the clock api doesn't take it well when two programs start to try to steer the same clock and may produce undesirable results for both in that case, so this doesn't seem horrible to me)